File: src/storage/src/main/java/sk/kasv/infinity/model/Video.java
package sk.kasv.infinity.model;

public class Video {

  Long id;
  Long courseId;
  String name;
  String description;

  public Video(Long id, Long courseId, String name, String description) {
    this.id = id;
    this.courseId = courseId;
    this.name = name;
    this.description = description;
  }

  public Video() {

  }

  public Long getId() {
    return id;
  }

  public Long getCourseId() {
    return courseId;
  }

  public String getName() {
    return name;
  }

  public String getDescription() {
    return description;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public void setCourseId(Long courseId) {
    this.courseId = courseId;
  }

  public void setName(String name) {
    this.name = name;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  @Override
  public String toString() {
    return "Video{" +
      "id=" + id +
      ", courseId=" + courseId +
      ", name='" + name + '\'' +
      ", description='" + description + '\'' +
      '}';
  }
}

File: src/entities/video.ts
export class Video {
    public static clone(v: Video):Video{
        return new Video(v.id, v.courseId, v.name, v.description);
    }

    constructor(
        public id: number,
        public courseId: number,
        public name: string,
        public description: string
    ){}

    public toString():string {
        return `id: ${this.id}, courseId: ${this.courseId}, name: ${this.name}, description: ${this.description}`;

    }
}